pennylane/beta/plugins/default_tensor.py

```diff
@@ -400,36 +444,37 @@ def _add_node(self, A, wires, name="UnnamedNode"):
             node = A
         else:
             node = tn.Node(A, name=name, backend=self.backend)
-        self._nodes.append(node)
+
+        if key not in self._nodes:
+            self._nodes[key] = []
+        self._nodes[key].append(node)
+
         return node
 
     def _add_edge(self, node1, idx1, node2, idx2):
-        """Adds an edge to the underlying tensor network.
-
-        The edge is also added to ``self._edges`` for bookkeeping.
+        """Adds an edge between two nodes.
 
         Args:
             node1 (tn.Node): first node to connect
-            idx1 (int): index of node1 to add the edge to
+            idx1 (int): index of node1 to connect the edge to
             node2 (tn.Node): second node to connect
-            idx2 (int): index of node2 to add the edge to
+            idx2 (int): index of node2 to connect the edge to
 
         Returns:
             tn.Edge: the newly created edge
         """
         edge = tn.connect(node1[idx1], node2[idx2])
-        self._edges.append(edge)
-
         return edge
 
-    def pre_apply(self):
-        self.reset()
-
     def apply(self, operation, wires, par):
         if operation == "QubitStateVector":
             state = self._array(par[0], dtype=self.C_DTYPE)
             if state.ndim == 1 and state.shape[0] == 2 ** self.num_wires:
-                self._state_node.tensor = self._reshape(state, [2] * self.num_wires)
+                tensor = self._reshape(state, [2] * self.num_wires)
+                self._clear_network()
+                self._add_initial_state_nodes(
+                    [tensor], [list(range(self.num_wires))], ["QubitStateVector"]
```

Comment:
Or comparable along the lines of including the `wires` in the name
```suggestion
                    [tensor], [list(range(self.num_wires))], ["QubitStateVector".format(tuple(wires))]
```

Reply:
Thanks for suggestion. Since the name is purely used for debugging right now, I've decided not to spend the effort to update this at the moment, but we could revisit in the future.